src/lib/marshal/translation/reader.ts

```diff
@@ -48,11 +53,13 @@ const readTranslationFiles = async (
     if (namespace === SYSTEM_NAMESPACE) continue;
 
     // eslint-disable-next-line no-await-in-loop
-    const [content, readJsonErrors] = await readJson(abspath);
+    const [content, sourceError, format] = await readTranslationFile(abspath);
+    if (sourceError) {
+      errors.push(sourceError);
+      continue;
+    }
 
-    if (readJsonErrors.length > 0) {
-      const e = new SourceError(formatErrors(readJsonErrors), abspath);
-      errors.push(e);
+    if (!content) {
       continue;
     }
```

Comment:
```suggestion
```
I'm not sure if we need this? My expectation for `readTranslationFile()` (and other read* functions) is that it should either return content and no error (undefined), or no content (undefined) and error(s). And since we have `if (sourceError) {}` above, I now expect `content` to always exist.  

I know we haven't fully typed the return value of `readTranslationFile()` like this (i.e. content and error being mutually exclusive), and we could and bake this expectation at the type level if you want to give it a go. I thought about suggesting it in my previous review, but didn't want to make too much fuss about it. Will leave this up to you.